### PR TITLE
fix(forms): indicate that the field is mandatory

### PR DESCRIPTION
**Pull request by jare-talend**
**Please check if the PR fulfills these requirements**

- [X] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
When we indacate that a field is mandatory, * is missing on the label

**What is the new behavior?**
Add * on the label if the field is required

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://246.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://246.talend.surge.sh)

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -124,7 +128,7 @@ The react/require-extension rule is deprecated. Please use the import/extensions
    30:18  error  'formData' is missing in props validation  react/prop-types
   162:14  error  'onChange' is missing in props validation  react/prop-types
 
-✖ 110 problems (110 errors, 0 warnings)
+✖ 112 problems (112 errors, 0 warnings)
```
Comment (jlame) [inline output/forms.eslint.txt]: 2 lines lint to fix

File: packages/forms/src/templates/FieldTemplate.js [hunk 2]
```diff
@@ -38,7 +39,7 @@ const CustomFieldTemplate = ({
 			<Label className="form-label" label={label} required={required} id={id} />}
 			{children}
 			{!hasLabelBefore && !isToggle && displayLabel &&
-			<Label label={label} required={required} id={id} className="control-label" />}
+			<Label label={label} required={required || schema.required} id={id} className="control-label" />}
```
Comment (jlame) [inline packages/forms/src/templates/FieldTemplate.js]: not sure to understand the difference between 
`"fullDatalist": {
       "required": true,
       "$ref": "#/definitions/datalist"
     }`

and `"required": [
      "select1"
    ]`

Comment (jlame) [inline packages/forms/src/templates/FieldTemplate.js]: Discussed with @jare-talend, the array notation is used in schema-form v4. There's an issue on required field validation, this PR is a workaround for this issue but the required in property form was used in v3.
We should wait for the validation issue to be fixed in v4, so that we don't make unnecessary change in backend.

Comment (jlame) [inline packages/forms/src/templates/FieldTemplate.js]: https://github.com/mozilla-services/react-jsonschema-form/issues/384

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://246.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://246.talend.surge.sh)
